src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -0,0 +1,139 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.RepositoryType;
+import com.google.sps.placeGuide.PlaceGuide;
+import com.google.sps.placeGuide.repository.PlaceGuideRepository;
+import com.google.sps.placeGuide.repository.PlaceGuideRepositoryFactory;
+import com.google.appengine.api.datastore.GeoPt;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.IllegalStateException;
+import java.io.IOException;
+
+/**
+ * This servlet handles placeguide's data.
+ */
+@WebServlet("/place-guide-data")
+public class PlaceGuideServlet extends HttpServlet {
+  
+  private final String userId;
+
+  // For production.
+  public PlaceGuideServlet() {
+    userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
```

Comment:
this(UserServiceFactory.getUserService().getCurrentUser().getUserId());

Reply:
done.